test/src/main/java/org/springframework/security/test/web/servlet/setup/SecurityMockMvcConfigurer.java

```diff
@@ -123,33 +123,47 @@ Filter getDelegate() {
 
 		@Override
 		public void init(FilterConfig filterConfig) throws ServletException {
+			this.requireDelegate();
 			this.delegate.init(filterConfig);
 		}
 
 		@Override
 		public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
 				throws IOException, ServletException {
+			this.requireDelegate();
 			this.delegate.doFilter(request, response, chain);
 		}
 
 		@Override
 		public void destroy() {
+			this.requireDelegate();
 			this.delegate.destroy();
 		}
 
 		@Override
 		public int hashCode() {
+			this.requireDelegate();
 			return this.delegate.hashCode();
 		}
 
 		@Override
 		public boolean equals(Object obj) {
+			this.requireDelegate();
 			return this.delegate.equals(obj);
 		}
 
 		@Override
 		public String toString() {
+			this.requireDelegate();
```

Comment:
We don't use `this.` on methods...only members.

Reply:
Okay, I fixed it.
If you think I could send another PR with a new checkstyle rule, something like this:

`<module name="Regexp">
			<property name="format" value="this\s*\.[a-zA-Z]+\(\)"/>
			<property name="illegalPattern" value="true"/>
			<property name="message" value="this. on method"/>
		</module>`

Off course this PR would include the fixing of existing matches on that pattern as well (currently there are 100+)

Reply:
That would be great. Thanks!